### ISSUE TITLE
didn't learn

### ISSUE THREAD

**Issue by CXY573** [meijieru/crnn.pytorch#107]
hey i make the lmdb using the create_dataset.py
and using it to train the model crnn_main.py
but i got something wrong
Start val
-------------------------e => e                   , gt: inquiry             
-------------------------e => e                   , gt: tendered            
-------------------------e => e                   , gt: insincerity         
-------------------------e => e                   , gt: conveyor            
-------------------------e => e                   , gt: neater              
-------------------------e => e                   , gt: glamour             
-------------------------e => e                   , gt: gurney              
-------------------------e => e                   , gt: subtle              
-------------------------e => e                   , gt: uaw                 
-------------------------e => e                   , gt: impoliteness        
Test loss: inf, accuray: 0.000000

**Comment by CXY573** [comment]
it seems that the net didn't work

**Comment by meijieru** [comment]
This might help you #92

**Comment by CXY573** [comment]
@meijieru  Thank you for apply.I have change my learning_rate to see whether it can work.
But i got some problem in reading code in crnn_main.py

image = torch.FloatTensor(opt.batchSize, 3, opt.imgH, opt.imgH)
I think it should be [batc_size,chanel,image_high,imag_wride]
Or it is just like a sliding windows with the shape(32,32)?

and text = torch.IntTensor(opt.batchSize * 5)
why the text have the size of 5*batchSize ,what is the meaning of 5?

**Comment by CXY573** [comment]
@1711120070 Sorry,I am not familiar with lmdb.I just follow the 
https://github.com/bgshih/crnn/tree/master/tool
But since i have not succeed yet,so i have no idea if i have maked the lmdb in the right way.

**Comment by 1711120070** [comment]
@CXY573 def writeCache(env, cache):
    with env.begin(write=True) as txn:
        for k, v in cache.iteritems():
            txn.put(k, v)
I run this, it reported error. So I change txn.put(k, v) to txn.put(k.encode(), v.encode()). Are you same with me?

**Comment by CXY573** [comment]
@meijieru 
and text = torch.IntTensor(opt.batchSize * 5)
why the text have the size of 5*batchSize ,what is the meaning of 5?

**Comment by CXY573** [comment]
@1711120070  You must make a mistake of using python3 to run the code which should be run with python 2,change your version you will make it

**Comment by 1711120070** [comment]
@CXY573 could you tell me how you modify https://github.com/bgshih/crnn/tree/master/tool.
run this directly would appear error.

**Comment by 1711120070** [comment]
@CXY573 thank you, i will have a try

**Comment by cddlyf** [comment]
@CXY573 your training problem solved now? just by changing the python version? With python 3, I just change the dataset convert code like @1711120070, and he lmdb can be generated. During training, I also have visualized the image and grountruth, it should be correct. But when training from scratch, the training output is like what you posted in the first slice. I noticed that the loss even becomes "inf" for some iterations.

**Comment by cddlyf** [comment]
@CXY573 @1711120070 By just decreasing the learning rate, the network looks converging now.  It is not the python version problem.
